drivers/syscon/Kconfig

```diff
@@ -0,0 +1,38 @@
+# SYSCON configuration options
+
+# Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# SYSCON options
+#
+menuconfig SYSCON
+	bool "SYSCON (System Controller) drivers"
+	help
```

Comment:
I think we should add:

`depends on MMU`

Since that is the main use case we are saying for syscon.

Reply:
why? there is no dependency on MMU at all.

Reply:
I understand, but isn't the main purpose of syscon for MMU SoC.  For the non-MMU SoC we are utilizing direct access of the registers.

Reply:
a platform without MMU can still define a syscon device in the DT (plenty of examples in the linux repo) and use this driver to access it. Adding a dependency on MMU makes no sense to me.